### PR TITLE
test: add guard training pipeline coverage

### LINKED ISSUE #239: Create Standardized ML Training Pipeline for Safety Prediction System
## Description
The current project lacks a structured and reproducible ML workflow for training and evaluating safety prediction models.

This issue proposes building a standardized machine learning pipeline to improve:
- reproducibility
- maintainability
- scalability
- model evaluation consistency

### 1. Dataset Organization
- Structured dataset folders
- Version-friendly format
- CSV validation checks

### 2. Data Preprocessing
- Missing value handling
- Feature encoding
- Feature scaling
- Train/test split pipeline

### 3. Training Pipeline
- Modular training scripts
- Configurable model selection
- Hyperparameter support

### 4. Evaluation System
- Accuracy
- Precision
- Recall
- F1-score
- Confusion matrix visualization

### 5. Model Persistence
- Save trained models using joblib/pickle
- Standardized inference loading

### 6. Inference API
- REST API endpoint for predictions
- JSON-based input/output

### 7. Documentation
- Setup instructions
- Training guide
- API usage examples

## Suggested Tech Stack
- Python
- Scikit-learn
- Pandas
- FastAPI / Flask
- Joblib

## Expected Outcome
A clean and reusable ML pipeline that allows contributors to:
- train models easily,
- evaluate performance consistently,
- deploy inference APIs reliably.

### PR DESCRIPTION
**Pull request by joicyroslin-svg**
Closes #239

### Summary

Adds focused test coverage for the Guard safety prediction training pipeline utilities.

### Changes Made

* Added tests for training dataset preprocessing.
* Verified missing/invalid dataset columns raise a clear error.
* Added config loading test to ensure YAML overrides merge correctly with default training settings.
* Covered label normalization, text cleanup, duplicate removal, and reproducibility-related config behavior.

### Testing

* `pytest backend/tests/test_guard_training_pipeline.py -v`

Result: 3 passed.

### CODE REVIEW

**Comment by joicyroslin-svg** [comment]
Hi @SdSarthak, I reviewed all failing CI checks for this PR.

This PR only adds one backend test file:

`backend/tests/test_guard_training_pipeline.py`

I verified the added test locally:

`pytest backend/tests/test_guard_training_pipeline.py -v`

Result: 3 passed.

Current CI failures appear unrelated to this PR:

* Backend Tests are failing in existing RAG/Auth tests such as `test_rag_ingest.py`, `test_rag_query.py`, `test_rag_stream.py`, `test_rag_feedback.py`, and `test_rag_pipeline.py`.
* Pytest auto-discover is also failing due to existing RAG integration tests returning 404 for RAG endpoints.
* Frontend Lint & Build is failing in existing frontend files like `Login.tsx`, `Register.tsx`, and `api.ts` because of unrelated lint/type issues.

Please let me know if you’d like any change in this PR, but the current failures seem outside the scope of the added guard training pipeline test coverage.
